server/src/main/java/org/apache/kafka/server/AssignmentsManager.java

```diff
@@ -93,10 +100,24 @@ public AssignmentsManager(Time time,
                 "broker-" + brokerId + "-directory-assignments-manager-",
                 new ShutdownEvent());
         channelManager.start();
+        this.metricsGroup.newGauge(QUEUE_REPLICA_TO_DIR_ASSIGNMENTS_METRIC_NAME, new Gauge<Integer>() {
+            @Override
+            public Integer value() {
+                return getMapSize(inflight) + getMapSize(pending);
```

Comment:
The map references are copied into `getMapSize` which avoids a check-then-act NPE race.